### PR TITLE
WEBP for QGIS Server

### PR DESCRIPTION
**Pull request by m-kuhn**
## Description

[Because](https://www.opengis.ch/2020/06/09/offline-wms-benchmarking-raster-formats-for-qfield/) [why not](https://developers.google.com/speed/webp/).

### CODE REVIEW

**Comment by andreasneumann** [comment]
Nice - how about the opposite? Consuming WebP as a WMS client in QGIS desktop?

Would that be equally easy to implement like enabling it in QGIS server?

**Comment by m-kuhn** [comment]
> Consuming WebP as a WMS client in QGIS desktop?

Good question. Likely yes. But I guess answering the question is best done through an implementation (which is not on my radar today)

**Review by elpaso — APPROVED** [review]
Nice!

**Review by rldhont — APPROVED** [review]

File: src/server/services/wms/qgswmsutils.cpp [hunk 1]
```diff
@@ -74,11 +74,16 @@ namespace QgsWms
     {
       return PNG;
     }
-    else if ( format.compare( QLatin1String( "jpg " ), Qt::CaseInsensitive ) == 0  ||
+    else if ( format.compare( QLatin1String( "jpg" ), Qt::CaseInsensitive ) == 0  ||
```
Comment (m-kuhn) [inline src/server/services/wms/qgswmsutils.cpp:77]: @elpaso @rldhont do you know what's this for? I wonder if that's actually ever used. It wasn't functional before (?) and nobody complained.

Comment (pblottiere) [reply src/server/services/wms/qgswmsutils.cpp:77]: I think that most people use `image/jpeg` (as well as unit tests) so it probably slipped under the radar.

Comment (elpaso) [reply src/server/services/wms/qgswmsutils.cpp:77]: I think it's been there for ages, It's not a standard mime type and untested. I believe the idea was just to be tolerant in what we accept.

**Comment by rldhont** [comment]
@m-kuhn thanks, just said just done, you have been faster then me.

**Comment by andreasneumann** [comment]
> which is not on my radar today

fair enough ;-)

**Review by sbrunner — APPROVED** [review]

File: src/server/services/wms/qgswmsutils.cpp (hunk 1 above)
Comment (m-kuhn) [reply src/server/services/wms/qgswmsutils.cpp:77]: What's your opinion, fix it, leave it or ditch it?
I think `image/jpg` or `jpeg` would be just as likely to be used, rendering this nothing but a random string.

Comment (rldhont) [reply src/server/services/wms/qgswmsutils.cpp:77]: fix it, isn't it ?

Comment (m-kuhn) [reply src/server/services/wms/qgswmsutils.cpp:77]: The code is a) broken b) non standard c) almost random and d) very likely unused. Four reasons for removing. I find it hard to find a good one for keeping.

Comment (m-kuhn) [reply src/server/services/wms/qgswmsutils.cpp:77]: What does "fixing" mean though? Simply removing the `' '` at the end of the string? Or harmonizing with the content type and changing `'jpg'` to `'jpeg'`? Or adding both? Isn't any of that breaking the interface, someone could already be sending requests with `jpg ` right now?

If it was really "broken code" I'd be all in for a fix. But I'm having a hard time seeing this as broken code. It looks more like a desperate attempt, pretending to be tolerant (as confirmed by @elpaso above).

Comment (elpaso) [reply src/server/services/wms/qgswmsutils.cpp:77]: @m-kuhn I think that in the old times it was more common for clients to use "jpg" "jpeg" with or without "image/", lower or upper case, I have no idea if that's still the situation today, I guess it's not.

So, if fixing does not cost too much I would leave that possibility, but if we do, we should document and put it under test. 

I don't really have a strong opinion here.

**Comment by m-kuhn** [comment]
Shall we still ship that with 3.14?
It's a super small patch, very low risk but hard to label "bugfix".

**Comment by elpaso** [comment]
Ship it!

**Comment by pblottiere** [comment]
+1 to merge :+1:

**Comment by m-kuhn** [comment]
:tada:

**Pull request merged**